sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/ConnectFailed.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Communication.CallAutomation
+{
+    /// <summary>
+    /// The Connect Failed event.
+    /// </summary>
+    public partial class ConnectFailed : CallAutomationEventBase
+    {
+        /// <summary>
+        /// Reason code.
+        /// </summary>
+        public MediaEventReasonCode ReasonCode { get; internal set; }
```

Comment:
Can you check if there is a generic Reason Code. I don't think this should be MediaEventReasonCode.